monkey/infection_monkey/monkey.py

```diff
@@ -256,8 +261,10 @@ def self_delete():
                                      close_fds=True, startupinfo=startupinfo)
                 else:
                     os.remove(sys.executable)
+                    T1107Telem(ScanStatus.USED, sys.executable).send()
```

Comment:
use a variable to determine whether deletion was successful or not. Right after the try-catch clause, send the telem according to result

Reply:
what happens if code goes to the delayed cmd section? We then need to set variable to false and after try-except we need to check if it's not false. I think it's better the way it is now.

Reply:
I disagree. even if the resulting code will be longer by 1-2 lines, it'll remove code duplication.
before the if-else statement, initialize a 'status' variable to `None`. Set it to USED or SCANNED in the relevant lines. After the if-else, send the telem (if status!=None)